src/main/java/training/endava/skeleton/MainApplication.java

```diff
@@ -1,64 +1,223 @@
+                .flatMap(List<Person>::stream)
+                .reduce(new Person(),(per1,per2)->{per1.setName(per1.getName()+per2.getName()); return per1;});
+
+
+
+
+        System.out.println("****************************************************************************");
+//        6. Transform a stream of companies in one of persons, in this way printing all the persons that are employed.
+
+        List<Person> listOfPersons=companiesList.stream()
+                .map(p -> p.getPersonList())
+                .flatMap(List<Person>::stream)
+                .collect(Collectors.toList());
+
+        System.out.println(listOfPersons);
+
+
+
+        System.out.println("****************************************************************************");
+//        7. Display the unique persons using distinct(). Two persons are considered to be identical if they have the same phone number.
+
+        List<Person> uniqueListOfPersons=companiesList.stream()
+                .map(p->p.getPersonList())
+                .flatMap(List<Person>::stream)
+                .distinct()
+                .collect(Collectors.toList());
+
+        System.out.println(uniqueListOfPersons);
+
+
+
+        System.out.println("****************************************************************************");
+//        8. Implement a custom collector and print the results of applying it on a stream.
+
+        Collector<Person, StringJoiner, String> customCollectorDefined=
+                Collector.of(
+                        ()->new StringJoiner(", " , "List of Persons: "," "),
+                        (supp,person) ->supp.add(person.getName().toUpperCase()),
+                        StringJoiner::merge,
+                        StringJoiner::toString
+                );
+        String listNamesOfPersons=persons.stream().collect(customCollectorDefined);
+        System.out.println(listNamesOfPersons);
+
+
+
+        System.out.println("****************************************************************************");
+//        9. Create a big stream (at least 1000 elements). Use stream and parallelStream and some processing operations. Measure the results (e.g. "Stream took x seconds / ParallelStream took y seconds").
+
+        List<Person> bigStream=new ArrayList<>();
+
+        for (int i=0;i<=1010;i++){
+            Person person=new Person(i, "Vasile", "075XXXXXX", "Address X");
+            bigStream.add(person);
+        }
+
+        long startStream=System.nanoTime();
+
+        bigStream.stream()
+                .sorted(Comparator.comparing(Person::getId).reversed())
+                .filter(i->i.getId()%2==0);
+
+
+        long endStream=System.nanoTime();
+        System.out.println("Stream: "  + (endStream-startStream));
+
+
+        startStream=System.nanoTime();
+
+        bigStream.parallelStream()
+                .sorted(Comparator.comparing(Person::getId).reversed())
+                .filter(i->i.getId()%2==0);
+
```

Comment:
Add the intermediate operation .peek(System.out::println) to your big streams. 
Do you see anything printed? Do you remember the laziness property of streams?